exporting/prometheus/prometheus.c

```diff
@@ -326,6 +326,47 @@ void format_host_labels_prometheus(struct instance *instance, RRDHOST *host)
     rrdlabels_walkthrough_read(host->rrdlabels, format_prometheus_label_callback, &tmp);
 }
 
+/**
+ * Check the input buffer for the label name and maybe remove the label.
+ *
+ * @param b the buffer to check.
+ * @param name the label name to search for.
+ * @param keepExistingLabel zero to delete the existing label, non-zero otherwise.
+ * @return Returns 1 if the label exists and is kept. Returns 0 in all other cases.
+ */
+static int check_existing_label(BUFFER *b, const char *name, int keepExistingLabel) {
+    char k[PROMETHEUS_ELEMENT_MAX + 2];
+    size_t len = prometheus_name_copy(k, name, PROMETHEUS_ELEMENT_MAX);
+    k[len++] = '=';
+    k[len] = '\0';
+    const char *end = b->buffer + b->len;
+    char *match = strstr(b->buffer, k);
+    if (match == NULL)
+        return 0;
+    else if (keepExistingLabel)
+        return 1;
+    const char *matchEnd = strchr(match, ',');
+    if (matchEnd != NULL) {
+        // Have subsequent label(s), shift them to discard the matched label.
+        ++matchEnd;
+        memmove(match, matchEnd, end - matchEnd);
+        match[end - matchEnd] = '\0';
+        b->len -= matchEnd - match;
+    } else if (unlikely(match == b->buffer)) {
+        // Match is the only label in the buffer, set length to 0.
```

Comment:
Hello @kevin-fwu ,

Your PR is working amazing and I understand you reach your goal, but I will give few suggestions to simplify future code maintenance. I suggest to use function [buffer_reset](https://github.com/netdata/netdata/blob/56499814dbbe6b6a82235f32e1c4ca2d142b0c28/libnetdata/buffer/buffer.h#L129)  instead to do this manually.

Reply:
This is in reference to the else if case? Yes, I can change this.